contracts/DomainRegistry.sol

```diff
@@ -23,94 +58,128 @@ contract DomainRegistry is Initializable, OwnableUpgradeable {
         if (level != uint8(1)) {
             string memory parentDomain = DomainParserLibrary.getParentDomain(domainName);
-            require(domains[parentDomain] != address(0), "Parent domain doesn't exist");
+            if (domains[parentDomain] == address(0)) {
+                revert DomainRegistryParentDomainNotExists({ domainName: domainName });
+            }
         }
         _;
     }
 
+    /// @notice Modifier to check if a domain is available for registration
+    /// @param domainName The domain name to check for availability
     modifier checkDomainAvailability(string memory domainName) {
-        require(domains[domainName] == address(0), "Domain is already reserved");
+        if (domains[domainName] != address(0)) {
+            revert DomainRegistryDomainIsAlreadyReserved({ domainName: domainName });
+        }
         _;
     }
 
+    /// @notice Modifier to check if the length of the domain name is valid
+    /// @param domainName The domain name to check the length for
     modifier checkDomainLength(string memory domainName) {
         bytes memory domainNameBytes = bytes(domainName);
 
-        require(domainNameBytes.length >= 2 && domainNameBytes.length <= 253, "Domain length should be between 2 and 253");
+        if (domainNameBytes.length < 2 || domainNameBytes.length > 253) {
+            revert DomainRegistryDomainNameLengthIsNotValid({ domainName: domainName });
+        }
         _;
     }
 
+    /// @notice Modifier to check if the sender is the owner of the domain when releasing it
+    /// @param domainName The domain name to check ownership for
     modifier checkDomainReleasing(string memory domainName) {
-        require(domains[domainName] != address(0), "Domain is not registered yet");
-        require(domains[domainName] == msg.sender, "Domain should be unregistered by the domain owner");
+        if (domains[domainName] == address(0)) {
+            revert DomainRegistryDomainIsNotRegistered({ domainName: domainName });
+        }
+        if (domains[domainName] != msg.sender) {
+            revert DomainRegistryOnlyDomainOwnerAllowed({ domainName: domainName });
+        }
         _;
     }
 
+    /// @notice Retrieves the owner of a domain
+    /// @param domainName The domain name to query the owner for
+    /// @return The owner's address of the domain
     function getDomain(string memory domainName) public view returns (address) {
         string memory rootDomain = DomainParserLibrary.getRootDomain(domainName);
 
         return domains[rootDomain];
     }
 
-    function validateDomainRegistration(string memory domainName) internal
-        checkDomainParent(domainName)
-        checkDomainAvailability(domainName)
-        checkSufficientEtn(domainName)
-    {
+    /// @notice Validates domain registration by checking availability, parent existence, and ETN sent
+    /// @param domainName The domain name to validate registration for
+    function validateDomainRegistration(
```

Comment:
It is better to underscore all the private and internal functions and state vars. If you do so, the convenience of first contract read will increase a lot.
https://docs.soliditylang.org/en/latest/style-guide.html#underscore-prefix-for-non-external-functions-and-variables


 -1 point

Reply:
done